roles/sap_netweaver_preconfigure/tasks/sapnote/assert-3119751.yml

```diff
@@ -5,46 +5,88 @@
+  ignore_errors: "{{ sap_netweaver_preconfigure_assert_ignore_errors | d(false) }}"
+
+- name: Set fact for the latest 'compat-sap-c++.NUM.so' symlink
+  ansible.builtin.set_fact:
+    __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest: >-
+     {{ __sap_netweaver_preconfigure_register_find_compat_sap_cpp.files
+       | map(attribute = 'path')
+       | sort | last | basename }}
+  when: __sap_netweaver_preconfigure_register_find_compat_sap_cpp.matched > 0
+
+# Verify /opt/rh/SAP/lib64/libstdc++.so.6
+- name: Get info about file '/opt/rh/SAP/lib64/libstdc++.so.6'
   ansible.builtin.stat:
-    path: /opt/rh/SAP/lib64/compat-sap-c++-10.so
-  register: __sap_netweaver_preconfigure_register_stat_compat_sap_cpp
+    path: /opt/rh/SAP/lib64/libstdc++.so.6
+  register: __sap_netweaver_preconfigure_register_stat_opt_rh_libstdc_assert
+  failed_when: false
 
-- name: Report if checking for a link from libstdc++.so.6 to compat-sap-c++-10.so is skipped
-  ansible.builtin.debug:
-    msg: "INFO: Not checking for link '{{ sap_netweaver_preconfigure_rpath }}/libstdc++.so.6' - file '/opt/rh/SAP/lib64/compat-sap-c++-10.so' does not exist on this system."
-  when: not __sap_netweaver_preconfigure_register_stat_compat_sap_cpp.stat.exists
+- name: Assert that file '/opt/rh/SAP/lib64/libstdc++.so.6' exists
+  ansible.builtin.assert:
+    that: __sap_netweaver_preconfigure_register_stat_opt_rh_libstdc_assert.stat.exists
+    fail_msg: "FAIL: File '/opt/rh/SAP/lib64/libstdc++.so.6' does not exist!"
+    success_msg: "PASS: File '/opt/rh/SAP/lib64/libstdc++.so.6' exists."
+  ignore_errors: "{{ sap_netweaver_preconfigure_assert_ignore_errors | d(false) }}"
+
+- name: Assert that file '/opt/rh/SAP/lib64/libstdc++.so.6' is a symlink
+  ansible.builtin.assert:
+    that: __sap_netweaver_preconfigure_register_stat_opt_rh_libstdc_assert.stat.islnk
+    fail_msg: "FAIL: File '/opt/rh/SAP/lib64/libstdc++.so.6' is not a symlink!"
+    success_msg: "PASS: File '/opt/rh/SAP/lib64/libstdc++.so.6' is a symlink."
+  ignore_errors: "{{ sap_netweaver_preconfigure_assert_ignore_errors | d(false) }}"
+  when: __sap_netweaver_preconfigure_register_stat_opt_rh_libstdc_assert.stat.exists
+
+- name: Assert that file '/opt/rh/SAP/lib64/libstdc++.so.6' is a symlink to '{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest }}'
+  ansible.builtin.assert:
+    that: __sap_netweaver_preconfigure_register_stat_opt_rh_libstdc_assert.stat.lnk_target == __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest
+    fail_msg: "FAIL: File '/opt/rh/SAP/lib64/libstdc++.so.6' is not a symlink to '{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest }}!'"
+    success_msg: "PASS: File '/opt/rh/SAP/lib64/libstdc++.so.6' is a symlink to '{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest }}'."
+  ignore_errors: "{{ sap_netweaver_preconfigure_assert_ignore_errors | d(false) }}"
+  when: __sap_netweaver_preconfigure_register_stat_opt_rh_libstdc_assert.stat.exists
```

Comment:
I would actually combine these 2 checks. Is there really much value in separately displaying that libstdc++.so.6 is a symlink, compared to a combined output in the second task like:
- fail: is not a symlink or not pointing to `<compat-sap-cpp-lastest>`
- pass: is a symlink pointing to `<compat-sap-cpp-latest>`

Reply:
Suggestion:

```
- name: Assert that file '/opt/rh/SAP/lib64/libstdc++.so.6' is a symlink to '{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest }}'
  ansible.builtin.assert:
    that:
      - __sap_netweaver_preconfigure_register_stat_opt_rh_libstdc_assert.stat.exists
      - __sap_netweaver_preconfigure_register_stat_opt_rh_libstdc_assert.stat.islnk
      - __sap_netweaver_preconfigure_register_stat_opt_rh_libstdc_assert.stat.lnk_target == __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest
    fail_msg: "FAIL: File '/opt/rh/SAP/lib64/libstdc++.so.6' does not exist or is not a symlink to '{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest }}!'"
    success_msg: "PASS: File '/opt/rh/SAP/lib64/libstdc++.so.6' is a symlink to '{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest }}'."
  ignore_errors: "{{ sap_netweaver_preconfigure_assert_ignore_errors | d(false) }}"

Reply:
Great suggestion, thanks!